mobify/platform-scaffold
# Wrap triggerAppStartEvent in setTimeout

A tracking pixel that's downloaded by our PWA was delaying a few other scripts from downloading for a short time (~300ms), so this PR is meant to slightly alter the timing our scripts trigger the pixel to download to a point in time that does not delay those scripts from downloading.

**JIRA**: [WEB-1441](https://mobify.atlassian.net/browse/WEB-1441)

## Changes
- Wrap the `triggerAppStartEvent` functino in a `setTimeout`

## Todos
- ~~(if applicable) analytics events instrumented to measure impact of change~~

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- **Verify** that the tracking pixel still downloads, i.e. by checking in Chrome's network panel. You can filter by `s.gif`

## Result

| Before | ![tracker-gif-present](https://user-images.githubusercontent.com/734535/27614639-cd78d37c-5b57-11e7-8916-fd0751d89533.png) |
| --- | --- |
| After | ![settimeout](https://user-images.githubusercontent.com/734535/27614644-d1f8f0b2-5b57-11e7-9f60-1f425d952cc3.png) |
